__tests__/app.test.js

```diff
@@ -38,12 +35,45 @@ describe("/api/topics", () => {
           },
         ]);
       });
-  });
-  // describe('invalid url error', () => {
-  //   test('should return correct error and message if invalid url used', () => {
-  //     .then((response) => {
-  //       expect(response.body.msg).toBe("Invalid url")
-  //     })
-  //   });
-  // });
+    });
 });
+describe("/api", () => {
+  test("GET: 200: responds with object containing all the ap endpoints", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then((response) => {
+        const { endpoints } = response.body;
+        expect(endpoints).toEqual({
+          "GET /api": {
+            description:
+              "serves up a json representation of all the available endpoints of the api",
+          },
+          "GET /api/topics": {
+            description: "serves an array of all topics",
+            queries: [],
+            exampleResponse: {
+              topics: [{ slug: "football", description: "Footie!" }],
+            },
+          },
+          "GET /api/articles": {
+            description: "serves an array of all articles",
+            queries: ["author", "topic", "sort_by", "order"],
+            exampleResponse: {
+              articles: [
+                {
+                  title: "Seafood substitutions are increasing",
+                  topic: "cooking",
+                  author: "weegembump",
+                  body: "Text from the article..",
+                  created_at: "2018-05-30T15:59:13.341Z",
+                  votes: 0,
+                  comment_count: 6,
+                },
+              ],
+            },
+          },
+        });
+      });
+    })
+  })
```

Comment:
This test works, for now, but you'd have to update it every endpoint. A more concise way to test this is by requiring in the endpoints file directly, then doing a 'toEqual' on the endpoints response. By requiring in the file, you'd avoid having to manually update it.

(this same requiring method can be used in the controller as well - because endpoints.json is a JSON file